### ISSUE TITLE
Ask for the script folder

### ISSUE THREAD

**Issue by hongge831** [WangYZ1608/Self-Prompt-Tuning#1]
Hi, nice work. Would you mind uploading the scripts foler? It seems that current codebase does not contain the foler.

**Comment by hongge831** [comment]
and would you please provide the train & test text list for us?

**Comment by WangYZ1608** [comment]
I tried uploading the scripts folder, but it was not successful because the file was large. I am trying other ways, such as through Baidu Cloud.

**Comment by hongge831** [comment]
> I tried uploading the scripts folder, but it was not successful because the file was large. I am trying other ways, such as through Baidu Cloud.

哈喽，能不能联系聊一下呀？

**Comment by WangYZ1608** [comment]
If you have any questions, please ask directly and we will answer them promptly.

**Comment by hongge831** [comment]
> If you have any questions, please ask directly and we will answer them promptly.

OK，here are my questions, and looking forward your detiled reply.
1. Are the hyperparameters different for different tasks (e.g. the lr and wd are different from CUB and CAR)? If yes, can you provide training scripts for different tasks? That is the scripts folder mentioned before.
2. I noticed that in the comparison experiment, you directly quoted the results of VPT. Have you tried randomly initializing the prompt to reproduce the vpt experiment? I mean, is it possible that the improvement in experimental results is due to other different hyperparameters (such as learning rate and optimizer) rather than prompt initialization? Do you have any experimental results for this part?
3. About the prompt construction time. I tried your code and found that the prompt construction time is much longer than mentioned in the paper (random sampling strategy). What is the reason?

**Comment by hongge831** [comment]
by the way, I try to reproduce the results of CUB (using the IN-21k sup backbone), I only got the results of 88.93% vs 90.6%（your paper Table4.b random sampling）, the hyper-param I use is the same as your readme. can you give me some suggestions?

**Comment by WangYZ1608** [comment]
> > If you have any questions, please ask directly and we will answer them promptly.
> 
> OK，here are my questions, and looking forward your detiled reply.
> 
> 1. Are the hyperparameters different for different tasks (e.g. the lr and wd are different from CUB and CAR)? If yes, can you provide training scripts for different tasks? That is the scripts folder mentioned before.
> 2. I noticed that in the comparison experiment, you directly quoted the results of VPT. Have you tried randomly initializing the prompt to reproduce the vpt experiment? I mean, is it possible that the improvement in experimental results is due to other different hyperparameters (such as learning rate and optimizer) rather than prompt initialization? Do you have any experimental results for this part?
> 3. About the prompt construction time. I tried your code and found that the prompt construction time is much longer than mentioned in the paper (random sampling strategy). What is the reason?

1. Different tasks require setting different hyper-params, including LR and WD.
2. For VPT，Gate-VPT，and other competitors, we quoted the results of the original paper and did not reproduce them. Hyper-parameter searches have been done in the original paper and/or official code of these methods.
3. Please check whether cuda is used.

**Comment by WangYZ1608** [comment]
![Snipaste_2024-02-20_11-51-33](https://github.com/WangYZ1608/Self-Prompt-Tuning/assets/100399172/9fb38e31-27be-42b7-bb5b-9dd5742d6fdf)
这是我刚做的结果，你可以参考下 （IN-21K supervised , CUB-200）

**Comment by hongge831** [comment]
> ![Snipaste_2024-02-20_11-51-33](https://private-user-images.githubusercontent.com/100399172/306128713-9fb38e31-27be-42b7-bb5b-9dd5742d6fdf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg0MTQwNzIsIm5iZiI6MTcwODQxMzc3MiwicGF0aCI6Ii8xMDAzOTkxNzIvMzA2MTI4NzEzLTlmYjM4ZTMxLTI3YmUtNDJiNy1iYjViLTlkZDU3NDJkNmZkZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMjIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDIyMFQwNzIyNTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNmE4MGQxMDBiM2VlMzNkMTkzZWJhNGRmOTUxOTNjYzljMjdkNzA3YTc3OTNkZTdhMWI5NjJmODVjMzlkODU2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.mexsIe7hvXa_kMXvndHMjLWDzTIOI3dLikNSFGXyLhg) 这是我刚做的结果，你可以参考下 （IN-21K supervised , CUB-200）

谢谢回复，还有如下两个问题想咨询一下：
1. 能否把FGVC这几个数据集的运行脚本上传一下呀？
2. 构建prompt的时候你的代码实现方式似乎没有用gpu，能否将您用gpu处理的代码上传一下呢？
3. 能否加微信交流学习下？我的邮箱是yeff@zju.edu.cn

**Comment by hongge831** [comment]
@WangYZ1608  hello？

**Comment by Noahsark** [comment]
I have the same issue. I also get only 88.9 instead of 90.6 (reported in the paper) following Table 4.b of random sampling, mean pooling, and max pooling on CUB. As a paper that is claimed to beat the previous SOTA and will become a new SOTA once it passes peer review, it is crucial to confirm the results are reproducible. Otherwise, it will cause confusion and more difficulties for the community.

We will try to report this issue to the conference once it passes peer review as a publication.

**Comment by WangYZ1608** [comment]
In the README, the recipe of sup (IN-21K) is used on Dogs. However, you seem to be using it for CUB-200, which of course will lead to different results from the paper.

We report mean results with 3 runs in the paper, and also provide some training curves as reference.

If you encounter problems when reproducing, please feel free to contact us at any time.

**Comment by WangYZ1608** [comment]
> I have the same issue. I also get only 88.9 instead of 90.6 (reported in the paper) following Table 4.b of random sampling, mean pooling, and max pooling on CUB. As a paper that is claimed to beat the previous SOTA and will become a new SOTA once it passes peer review, it is crucial to confirm the results are reproducible. Otherwise, it will cause confusion and more difficulties for the community.

In the README, the recipe of sup (IN-21K) is used on Dogs. However, you seem to be using it for CUB-200, which of course will lead to different results from the paper.

We report mean results with 3 runs in the paper, and also provide some training curves as reference.

If you encounter problems when reproducing, please feel free to contact us at any time.

**Comment by Noahsark** [comment]
@WangYZ1608 Thanks for the reply. Will you release the settings of ViT IN-21k for other datasets of FGVC to reproduce the results? Will you release the train/val splitting files for all datasets?

**Comment by hongge831** [comment]
@WangYZ1608 If you see the message, please post the details of FGVC’s reproduction. Thank you.

**Comment by TARTAMIN99** [comment]
> I have the same issue. I also get only 88.9 instead of 90.6 (reported in the paper) following Table 4.b of random sampling, mean pooling, and max pooling on CUB. As a paper that is claimed to beat the previous SOTA and will become a new SOTA once it passes peer review, it is crucial to confirm the results are reproducible. Otherwise, it will cause confusion and more difficulties for the community.

I try to reproduce the results of CUB (using the IN-21k sup backbone), I only got the results of 85.8%，Will you release the settings of ViT IN-21k for CUB200. Thanks for your help
